packages/nimble-components/src/combobox/testing/combobox.pageobject.ts

```diff
@@ -0,0 +1,125 @@
+import { keyEnter, keyArrowDown } from '@microsoft/fast-web-utilities';
+import type { Combobox } from '..';
+import { waitForUpdatesAsync } from '../../testing/async-helpers';
+import {
+    createEventListener,
+    waitAnimationFrame
+} from '../../utilities/tests/component';
+
+/**
+ * Page object for the `nimble-combobox` component to provide consistent ways
+ * of querying and interacting with the component during tests.
+ */
+export class ComboboxPageObject {
+    private readonly regionLoadedListener = createEventListener(
+        this.comboboxElement,
+        'loaded'
+    );
+
+    public constructor(protected readonly comboboxElement: Combobox) {}
+
+    public async waitForAnchoredRegionLoaded(): Promise<void> {
+        await this.regionLoadedListener.promise;
+    }
+
+    // The Selection API currently isn't supported properly or consistently in shadow DOM,
+    // so unfortunately our delete API has to be passed the post-deletion text.
+    public setInputText(text: string, asDelete = false): void {
+        this.comboboxElement.control.value = text;
+        const inputEvent = new InputEvent('input', {
+            data: text,
+            inputType: asDelete ? 'deleteContentForward' : 'insertText'
+        });
+        this.comboboxElement.control.dispatchEvent(inputEvent);
+    }
+
+    public commitValue(text: string): void {
```

Comment:
Perhaps add some docs that state that calling this method will result in a `change` event being emitted?